packages/indexer/src/sinks/drizzle/drizzle.ts

```diff
@@ -45,11 +69,17 @@ export class DrizzleSink<
 
   async transaction(
     cb: (params: {
-      db: PgTransaction<TQueryResult, TFullSchema, TSchema>;
+      transaction: (
+        endCursor?: Cursor,
+      ) => DrizzleSinkTransaction<TQueryResult, TFullSchema, TSchema>;
     }) => Promise<void>,
   ): Promise<void> {
     await this._db.transaction(async (db) => {
-      await cb({ db });
+      const transaction = (endCursor?: Cursor) => {
+        return new DrizzleSinkTransaction(db, endCursor);
+      };
+
+      await cb({ transaction });
```

Comment:
```suggestion
      const txn = new DrizzleSinkTransaction(db, endCursor);

      await cb({ db: txn });
```

User should not have to know it's a transaction. Let's pass it named `db`.